### ISSUE TITLE
[openmvs] build failure

### ISSUE THREAD

**Issue by ChickenTarm** [microsoft/vcpkg#40050]
### Operating system

Ubuntu 22.04

### Steps to reproduce the behavior

```Shell
`vcpkg install openmvs`
```

### Failure logs

```
[DEBUG] 1000: execute_process(curl --fail -L https://gitlab.gnome.org/GNOME/gtk/-/archive/3.24.38/gtk-3.24.38.tar.gz --create-dirs --output /home/vcpkg/downloads/GNOME-gtk-3.24.38.tar.gz.350643.part)
[DEBUG] 1000: cmd_execute_and_stream_data() returned 22 after 30063734 us
error: Missing GNOME-gtk-3.24.38.tar.gz and downloads are blocked by x-block-origin.
error: https://gitlab.gnome.org/GNOME/gtk/-/archive/3.24.38/gtk-3.24.38.tar.gz: curl failed to download with exit code 22
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

curl: (22) The requested URL returned error: 504

[DEBUG] /source/src/vcpkg/base/downloads.cpp(1030): 
[DEBUG] Time in subprocesses: 30063734us
[DEBUG] Time in parsing JSON: 1us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 194us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 30 s (30064124us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.
      
      3. Your proxy's remote server is out of service.
      
      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.
      

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_gitlab.cmake:113 (vcpkg_download_distfile)
  ports/gtk3/portfile.cmake:9 (vcpkg_from_gitlab)
  scripts/ports.cmake:192 (include)


error: building gtk3:x64-linux failed with: BUILD_FAILEDSee https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle gtk3:x64-linux: 30 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gtk3
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[gtk3]+Build+error+on+x64-linux&body=Copy+issue+body+from+%2Fhome%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```

### Additional context

For repo sources: gdk-pixbuf, pango, and gtk
The download phase of the install results in 504. I don't know if the recent outage affected them but it would be great to have fall back sources when the primary fails. In this case GNOME keeps a read only github that mirrors their gitlab.

So when 

`curl --fail -L https://gitlab.gnome.org/GNOME/gtk/-/archive/3.24.38/gtk-3.24.38.tar.gz --create-dirs --output /home/vcpkg/downloads/GNOME-gtk-3.24.38.tar.gz`

fails it then tries

`curl --fail -L https://github.com/GNOME/gtk/archive/refs/tags/3.24.38.tar.gz --create-dirs --output /home/vcpkg/downloads/GNOME-gtk-3.24.38.tar.gz`

**Comment by jimwang118** [comment]
Check the error message and it turns out that an error occurred during gtk download. Please ensure that the network is unobstructed and confirm that the gtk download website can be opened normally.

**Comment by ChickenTarm** [comment]
> Check the error message and it turns out that an error occurred during gtk download. Please ensure that the network is unobstructed and confirm that the gtk download website can be opened normally.

I wasn't able to access it before. Now it works and is accessible, but i guess my point still stands. If there are multiple sources available, it would be great to be able to use the fallbacks until all fallbacks have been exhausted instead of just giving up on the first failure.

**Comment by jimwang118** [comment]
> I wasn't able to access it before. Now it works and is accessible, but i guess my point still stands. If there are multiple sources available, it would be great to be able to use the fallbacks until all fallbacks have been exhausted instead of just giving up on the first failure.

We only obtain source code from the upstream official website, and other sources cannot guarantee the uniformity of the code.

**Comment by jimwang118** [comment]
Thanks for posting this issue. Please reopen this issue if this is still a problem for you.
